src/algorithms/WState.cpp

```diff
@@ -0,0 +1,26 @@
+#include "algorithms/WState.hpp"
+
+namespace qc {
+void fGate(QuantumComputation& qc, std::size_t i, std::size_t j, std::size_t n,
+           std::size_t k) {
```

Comment:
```suggestion
void fGate(QuantumComputation& qc, const std::size_t i, const std::size_t j, const std::size_t n,
           const std::size_t k) {
```
Small nitpick: could we sort the parameters alphabetically?